pkg/bloomgateway/client.go

```diff
@@ -109,9 +116,25 @@ func (i *ClientConfig) RegisterFlags(f *flag.FlagSet) {
 // RegisterFlagsWithPrefix registers flags for the Bloom Gateway client configuration with a common prefix.
 func (i *ClientConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	i.GRPCClientConfig.RegisterFlagsWithPrefix(prefix+"grpc", f)
+	i.Cache.RegisterFlagsWithPrefix(f, prefix+"cache.")
```

Comment:
This is a bit of a trap:

You call `RegisterFlagsWithPrefix` on the inlined object `resultscache.Config`. Therefore not all flags are registered.

Reply:
Fixed by implementing `RegisterFlagsWithPrefix` in `CacheConfig`